clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1581,7 +1581,27 @@ public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, Of
      * Overrides the fetch offsets that the consumer will use on the next {@link #poll(Duration) poll(timeout)}. If this API
      * is invoked for the same partition more than once, the latest offset will be used on the next poll(). Note that
      * you may lose data if this API is arbitrarily used in the middle of consumption, to reset the fetch offsets
+     * <p>
+     * The next Consumer Record which will be retrieved when poll() is invoked will have the offset specified, given that
+     * a record with that offset exists (ie: it is a valid offset).
+     * <p>
+     * {@link KafkaConsumer#seekToBeginning(Collection)} will go to the first offset in the topic.
+     * seek(0) is equivalent to seekToBeginning for a topic with beginning offset 0,
+     * assuming that there is a record at offset 0 still available.
+     * {@link KafkaConsumer#seekToEnd(Collection)} is equivalent to seeking to the highest known offset + 1.
```

Comment:
nit: `{@link #seekToEnd(Collection)}`

Reply:
is equivalent to `seek` to the highest known offset + 1

Reply:
Could we say:
{seekToEnd(Collection)} is equivalent to seeking to `the highest known offset + 1` -> {seekToEnd(Collection)} is equivalent to seeking to `the last offset of the partition`. 
That's the wording in the javadoc in `seekToEnd` method.

Reply:
One thing to note here, the seekToEnd might seek to `log end offset` or `Last Stable Offset` depends on `read-committed` or `read-uncommitted`. So we need to be cautious for the wording here.